sw/host/opentitanlib/src/crypto/ecdsa.rs

```diff
@@ -115,6 +117,15 @@ impl EcdsaRawSignature {
         Ok(sig)
     }
 
+    pub fn read_from_file(path: &Path) -> Result<EcdsaRawSignature> {
+        let err = |e: std::io::Error| Error::ReadFailed {
+            file: path.to_owned(),
+            source: anyhow!(e),
+        };
+        let mut file = File::open(path).map_err(err)?;
```

Comment:
It would be considered idiomatic rust to stuff the closure expression into the `mapp_err` argument:
```
let mut file = File::open(path).map_err(|e| Error::ReadFailed {
    file: path.to_owned(),
    source: anyhow!(e),
})?;
```

One other note: since this isn't a trait implementation with specific error type constraints, you could also just let `?` auto-coerce the `io::Error` into `anyhow::Error`.  You could attach a context message to the error to be friendly:
```
let mut file = File::open(path).context(format!("Failed to read {path:?}"))?;
````

Reply:
Or use `with_context` to defer the string formatting to only the error case:
```
use anyhow::Context;
let mut file = File::open(path).with_context(|| format!("Failed to read {path:?}"))?;
```

Reply:
I went for the last option. Thanks!

Done.